### PR TITLE
Merging branch Selenium tests into SEED develop

### PR DESCRIPTION
**Pull request by rHorsey**
Adding Selenium to the SEED environment.
Adding initial end-to-end testing through test_firefox.py selenium test.
Adding template in test_api.py for api 'unit-ish' tests.
Changed seed/tasks.py file to standardize on a single cache format.
Changed seed/tasks.py file to allow tasks to work either synchronously or asynchronously.
Changed travis build config.
Changed test and untracked_local settings files.
Added id tags to a number of partials to allow for selenium testing.

### CODE REVIEW

File: seed/tests/api/data/covered-buildings-sample.csv [hunk 1]
```diff
@@ -1,513 +0,0 @@
-UBI,GBA,BLDGS,Address,Owner,City,State,Zip,Property Type,AYB_YearBuilt
```
Comment (nllong) [inline seed/tests/api/data/covered-buildings-sample.csv]: @rHorsey: was this file removed on purpose?

File: seed/views/main.py [hunk 2]
```diff
@@ -2017,7 +2020,7 @@ def progress(request):
     progress_key = json.loads(request.body).get('progress_key')
 
     if cache.get(progress_key):
-        result = cache.get(progress_key)
+        result = get_cache(progress_key)
```
Comment (nllong) [inline seed/views/main.py]: Is this correct? Aren't sin't `cache.get` the new method for getting the cache data?

File: seed/tests/api/data/covered-buildings-sample.csv (hunk 1 above)
Comment (rHorsey) [inline seed/tests/api/data/covered-buildings-sample.csv]: Yes. We already have the file in seed/tests/data, and that's where all the tests were pointing to.

File: seed/views/main.py (hunk 2 above)
Comment (rHorsey) [inline seed/views/main.py]: Interesting. In tasks.py all the commands used on caches are get_cache and set cache. It should be cache.get and cache.set when we get all the progress stuff classified.

Comment (rHorsey) [inline seed/views/main.py]: Alex confirmed that it should be get_cache.

**Pull request merged**